dotCMS/src/main/java/com/dotmarketing/util/importer/ImportResultConverter.java

```diff
@@ -51,238 +38,182 @@ private ImportResultConverter() {
+     * include warnings, errors, messages, counters, results, identifiers, and other categorized
+     * data.
+     * @throws IllegalArgumentException if the provided result is null.
      */
-    public static Map<String, List<String>> toLegacyFormat(ImportResult result) {
-        Map<String, List<String>> legacyResults = new HashMap<>();
+    public static HashMap<String, List<String>> toLegacyFormat(final ImportResult result,
+            final User user) {
 
-        // Convert messages to legacy format by type
-        Map<ValidationMessageType, List<String>> messagesByType =
-                convertMessagesToLegacyFormat(result.messages());
-
-        legacyResults.put(KEY_WARNINGS, messagesByType.get(ValidationMessageType.WARNING));
-        legacyResults.put(KEY_ERRORS, messagesByType.get(ValidationMessageType.ERROR));
-        legacyResults.put(KEY_MESSAGES, messagesByType.get(ValidationMessageType.INFO));
-
-        // Add counters
-        List<String> counters = new ArrayList<>();
-        ResultData data = result.data();
-        counters.add("linesread=" + result.fileInfo().totalRows());
-        counters.add("errors=" + messagesByType.get(ValidationMessageType.ERROR).size());
-        counters.add("newContent=" + data.summary().created());
-        counters.add("contentToUpdate=" + data.summary().updated());
-        legacyResults.put(KEY_COUNTERS, counters);
-
-        // Add results summary
-        List<String> results = new ArrayList<>();
-        results.add(data.summary().created() + " new \"" + data.summary().contentType()
-                + "\" were created");
-        if (data.summary().updated() > 0) {
-            results.add(data.summary().updated() + " \"" + data.summary().contentType() +
-                    "\" contentlets updated");
+        if (result == null) {
+            throw new IllegalArgumentException("Import result cannot be null");
         }
-        legacyResults.put("results", results);
-
-        // Add header validation info if present
-        addHeaderValidationToLegacy(result.fileInfo().headerInfo(), legacyResults);
-
-        return legacyResults;
-    }
 
-    /**
-     * Converts the structured ImportHeaderValidationResult to the legacy HashMap format. This
-     * method ensures backward compatibility by formatting the structured data into the string-based
-     * format expected by legacy code.
-     *
-     * @param validationResult The structured ImportHeaderValidationResult to convert
-     * @param legacyResults    The legacy format results map to update
-     * @return The number of errors found during header validation
-     */
-    public static int headerValidationResultsToLegacyMap(
-            final HeaderValidationResult validationResult,
-            final Map<String, List<String>> legacyResults) {
+        try {
+            final var preview = result.type().equals(OperationType.PREVIEW);
+
+            final HashMap<String, List<String>> legacyResults = new HashMap<>();
```

Comment:
it would be better to create a bean for this, this is very old fashion

Reply:
One of the requirements for the ticket was to leave the logic exactly as it currently is, that why for the "transformer" we are using the same format/logic the import had when it is called from the existing struts action.